data-es6.js

```diff
@@ -6,6 +6,24 @@ var temp = {};
 var flag = "flagged";
 var fallthrough = "needs-polyfill-or-native";
 
+var typescript = {
+    compiler: { 
+        val: true, 
+        note_id: "typescript-compiler-or-native", 
+        note_html: "This feature is supported through a compiler error, or when the <code>--target ES6</code> flag is set and a native ES6 host is used." 
```

Comment:
We would report the static error in both cases during compilation, but you would only get a runtime error if the host is ES6. If you don't need the level of detail I'm fine with just reporting `true`, which is what other compilers/transpilers seem to be doing.